### PR TITLE
fix: excessive rerendering on home screen

### PR DESCRIPTION
**Pull request by pwltr**
### Description

The Home screen and several components lower in the tree were rerendering way too many times because of bad selectors. Reselect's `createSelector` is not doing its job correctly so I'm adding a custom `createShallowEqualSelector` that does correct shallow equality checking needed for some larger selectors. Also we should not memoize selectors with `createSelector` by default. Should be assessed case by case whether it's needed. For most simple value lookups it is not needed.

### Linked Issues/Tasks

https://github.com/synonymdev/bitkit/issues/2052

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [ ] Unit test
- [x] No test

### QA Notes

Make sure components still rerender when they should.

### CODE REVIEW

File: src/AppOnboarded.tsx [hunk 1]
```diff
@@ -111,7 +111,8 @@ const AppOnboarded = (): ReactElement => {
 					});
 				}
 				dispatch(updateUi({ isOnline: true }));
-				updateExchangeRates();
+				// FIXME: this runs too often
+				// updateExchangeRates();
```
Comment (pwltr) [inline src/AppOnboarded.tsx:115]: This needs to be moved to another place. Not sure why but this fires even when connection status did not change.

**Review by limpbrains — APPROVED** [review]
This is strange. Maybe it was because we are mutating state where we should not

**Comment by pwltr** [comment]
> This is strange. Maybe it was because we are mutating state where we should not

I think this is purely about selectors. Docs are not very clear but looks like `createSelector` too only does strict equality `===` for inputs and outputs. For getting non-primitive types from selectors we need to do shallow equality specifically.

**Comment by catch-21** [comment]
Nothing to specifically test. See https://github.com/synonymdev/bitkit/issues/2052#issuecomment-2255707531

**Pull request merged**